core/src/apps/ethereum/sign_tx.py

```diff
@@ -22,22 +25,37 @@
 MAX_CHAIN_ID = 2147483629
 
 
+def format_access_list(access_list):
```

Comment:
In my opinion the complexity of writing these functions over just formatting the data correctly and letting the RLP code handle it is not worth it. But I can simplify the `format_access_list` function.

Reply:
You're trading RAM for complexity, and it isn't a good tradeoff for Trezor environment in this case. If the encoding was more complex, maybe, but in this case it's literally just writing number of items first, and that's done in `sign_*` itself anyway. OTOH your approach creates a list of lists (which allocates a bunch of items on top of what's already allocated from protobuf), only to throw it away after encoding.

Reply:
I think you might be underestimating the complexity. But no matter, I have pushed changes now that calculate the length and do the encoding without turning it into a list of lists. I would say it sacrifices readability and ease of understanding, but of course up to you!

Reply:
I see what you mean, but this is mainly because the RLP API is bad. I did a refactor in #1704:

```
def access_list_item_length(item: EthereumAccessList) -> int:
	address_length = rlp.length(bytes_from_address(item.address))
	keys_length = rlp.length(item.storage_keys)
	return rlp.header_length(address_length + keys_length) + address_length + keys_length

def access_list_length(access_list: list[EthereumAccessList]) -> int:
	payload_length = sum(access_list_item_length(i) for i in access_list)
	return rlp.header_length(payload_length) + payload_length

def write_access_list(w: Writer, access_list: list[EthereumAccessList]) -> None:
	payload_length = sum(access_list_item_length(i) for i in access_list)
	rlp.write_header(w, payload_length, rlp.LIST_HEADER_BYTE)
	for item in access_list:
		address_bytes = bytes_from_address(item.address)
		address_length = rlp.length(address_bytes)
		keys_length = rlp.length(item.storage_keys)
		rlp.write_header(w, address_length + keys_length, rlp.LIST_HEADER_BYTE)
		rlp.write(w, address_bytes)
		rlp.write(w, item.storage_keys)
```
the main drawback of this code is that we need to call `bytes_from_address` repeatedly, but I don't see a good way around that

Reply:
I can rebase and use that when it is merged 👍

Reply:
I had a look through this and let me be the first to say I am not well versed in this ecosystem, but would it be possible to pre process the bytes_from_address information once for each time in the access list as a means of creation of its list entry, so its available on the object itself. 

Would be curious what over head is carried through the workflow in ram vs the cost of processing it each time throughout.

Reply:
> so its available on the object itself

We can't add new properties to protobuf messages unfortunately (they're not regular python objects under the hood).
We _could_ create a list of results of bytes_from_address, so that `address_bytes[i] == bytes_from_address(access_list[i].address)`, but that seems like a needless complication for the amount of computation this will entail in practice.

Reply:
In the latest rebase I use your code. I haven't had a chance to test it as my dev environment is somehow broken. See my other comment.

Reply:
This works in latest! 😄